File: push/src/main/java/com/xiaomi/mipush/sdk/AssemblePushCollectionsManager.java
package com.xiaomi.mipush.sdk;

import android.content.Context;
import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.push.service.OnlineConfig;
import com.xiaomi.xmpush.thrift.ConfigKey;
import java.util.HashMap;
import java.util.Map;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/AssemblePushCollectionsManager.class */
public class AssemblePushCollectionsManager implements AbstractPushManager {
    private static volatile AssemblePushCollectionsManager sInstance;
    private PushConfiguration mConfiguration;
    private Context mContext;
    private boolean oldOCValue = false;
    private Map<AssemblePush, AbstractPushManager> mManagers = new HashMap<>();

    /* JADX INFO: renamed from: com.xiaomi.mipush.sdk.AssemblePushCollectionsManager$2, reason: invalid class name */
    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/AssemblePushCollectionsManager$2.class */
    static /* synthetic */ class AnonymousClass2 {
        static final /* synthetic */ int[] $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush;

        static {
            int[] iArr = new int[AssemblePush.values().length];
            $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush = iArr;
            try {
                iArr[AssemblePush.ASSEMBLE_PUSH_HUAWEI.ordinal()] = 1;
            } catch (NoSuchFieldError e) {
            }
            try {
                $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[AssemblePush.ASSEMBLE_PUSH_FCM.ordinal()] = 2;
            } catch (NoSuchFieldError e2) {
            }
            try {
                $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[AssemblePush.ASSEMBLE_PUSH_COS.ordinal()] = 3;
            } catch (NoSuchFieldError e3) {
            }
            try {
                $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[AssemblePush.ASSEMBLE_PUSH_FTOS.ordinal()] = 4;
            } catch (NoSuchFieldError e4) {
            }
        }
    }

    private AssemblePushCollectionsManager(Context context) {
        this.mContext = context.getApplicationContext();
    }

    public static AssemblePushCollectionsManager getInstance(Context context) {
        if (sInstance == null) {
            synchronized (AssemblePushCollectionsManager.class) {
                try {
                    if (sInstance == null) {
                        sInstance = new AssemblePushCollectionsManager(context);
                    }
                } catch (Throwable th) {
                    throw th;
                }
            }
        }
        return sInstance;
    }

    private void initAssemblePushManager() {
        AbstractPushManager manager;
        AbstractPushManager manager2;
        AbstractPushManager manager3;
        AbstractPushManager manager4;
        PushConfiguration pushConfiguration = this.mConfiguration;
        if (pushConfiguration != null) {
            if (pushConfiguration.getOpenHmsPush()) {
                StringBuilder sb = new StringBuilder();
                sb.append("ASSEMBLE_PUSH : ");
                sb.append(" HW user switch : " + this.mConfiguration.getOpenHmsPush() + " HW online switch : " + AssemblePushHelper.isOpenAssemblePushOnlineSwitch(this.mContext, AssemblePush.ASSEMBLE_PUSH_HUAWEI) + " HW isSupport : " + PhoneBrand.HUAWEI.equals(AssemblePushUtils.getPhoneBrand(this.mContext)));
                MyLog.w(sb.toString());
            }
            if (this.mConfiguration.getOpenHmsPush() && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(this.mContext, AssemblePush.ASSEMBLE_PUSH_HUAWEI) && PhoneBrand.HUAWEI.equals(AssemblePushUtils.getPhoneBrand(this.mContext))) {
                if (!contain(AssemblePush.ASSEMBLE_PUSH_HUAWEI)) {
                    addManager(AssemblePush.ASSEMBLE_PUSH_HUAWEI, PushManagerFactory.getManager(this.mContext, AssemblePush.ASSEMBLE_PUSH_HUAWEI));
                }
                MyLog.v("hw manager add to list");
            } else if (contain(AssemblePush.ASSEMBLE_PUSH_HUAWEI) && (manager = getManager(AssemblePush.ASSEMBLE_PUSH_HUAWEI)) != null) {
                removeManager(AssemblePush.ASSEMBLE_PUSH_HUAWEI);
                manager.unregister();
            }
            if (this.mConfiguration.getOpenFCMPush()) {
                StringBuilder sb2 = new StringBuilder();
                sb2.append("ASSEMBLE_PUSH : ");
                sb2.append(" FCM user switch : " + this.mConfiguration.getOpenFCMPush() + " FCM online switch : " + AssemblePushHelper.isOpenAssemblePushOnlineSwitch(this.mContext, AssemblePush.ASSEMBLE_PUSH_FCM) + " FCM isSupport : " + AssemblePushUtils.isGoogleServiceSatisfied(this.mContext));
                MyLog.w(sb2.toString());
            }
            if (this.mConfiguration.getOpenFCMPush() && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(this.mContext, AssemblePush.ASSEMBLE_PUSH_FCM) && AssemblePushUtils.isGoogleServiceSatisfied(this.mContext)) {
                if (!contain(AssemblePush.ASSEMBLE_PUSH_FCM)) {
                    addManager(AssemblePush.ASSEMBLE_PUSH_FCM, PushManagerFactory.getManager(this.mContext, AssemblePush.ASSEMBLE_PUSH_FCM));
                }
                MyLog.v("fcm manager add to list");
            } else if (contain(AssemblePush.ASSEMBLE_PUSH_FCM) && (manager2 = getManager(AssemblePush.ASSEMBLE_PUSH_FCM)) != null) {
                removeManager(AssemblePush.ASSEMBLE_PUSH_FCM);
                manager2.unregister();
            }
            if (this.mConfiguration.getOpenCOSPush()) {
                StringBuilder sb3 = new StringBuilder();
                sb3.append("ASSEMBLE_PUSH : ");
                sb3.append(" COS user switch : " + this.mConfiguration.getOpenCOSPush() + " COS online switch : " + AssemblePushHelper.isOpenAssemblePushOnlineSwitch(this.mContext, AssemblePush.ASSEMBLE_PUSH_COS) + " COS isSupport : " + AssemblePushUtils.isColorOSPushSupport(this.mContext));
                MyLog.w(sb3.toString());
            }
            if (this.mConfiguration.getOpenCOSPush() && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(this.mContext, AssemblePush.ASSEMBLE_PUSH_COS) && AssemblePushUtils.isColorOSPushSupport(this.mContext)) {
                addManager(AssemblePush.ASSEMBLE_PUSH_COS, PushManagerFactory.getManager(this.mContext, AssemblePush.ASSEMBLE_PUSH_COS));
            } else if (contain(AssemblePush.ASSEMBLE_PUSH_COS) && (manager3 = getManager(AssemblePush.ASSEMBLE_PUSH_COS)) != null) {
                removeManager(AssemblePush.ASSEMBLE_PUSH_COS);
                manager3.unregister();
            }
            if (this.mConfiguration.getOpenFTOSPush() && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(this.mContext, AssemblePush.ASSEMBLE_PUSH_FTOS) && AssemblePushUtils.isFunTouchOSPushSupport(this.mContext)) {
                addManager(AssemblePush.ASSEMBLE_PUSH_FTOS, PushManagerFactory.getManager(this.mContext, AssemblePush.ASSEMBLE_PUSH_FTOS));
            } else {
                if (!contain(AssemblePush.ASSEMBLE_PUSH_FTOS) || (manager4 = getManager(AssemblePush.ASSEMBLE_PUSH_FTOS)) == null) {
                    return;
                }
                removeManager(AssemblePush.ASSEMBLE_PUSH_FTOS);
                manager4.unregister();
            }
        }
    }

    public void addManager(AssemblePush assemblePush, AbstractPushManager abstractPushManager) {
        if (abstractPushManager != null) {
            if (this.mManagers.containsKey(assemblePush)) {
                this.mManagers.remove(assemblePush);
            }
            this.mManagers.put(assemblePush, abstractPushManager);
        }
    }

    public boolean contain(AssemblePush assemblePush) {
        return this.mManagers.containsKey(assemblePush);
    }

    public AbstractPushManager getManager(AssemblePush assemblePush) {
        return this.mManagers.get(assemblePush);
    }

    public boolean getUserSwitch(AssemblePush assemblePush) {
        boolean openHmsPush = false;
        switch (AnonymousClass2.$SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[assemblePush.ordinal()]) {
            case 1:
                PushConfiguration pushConfiguration = this.mConfiguration;
                openHmsPush = false;
                if (pushConfiguration != null) {
                    openHmsPush = pushConfiguration.getOpenHmsPush();
                }
                break;
            case 2:
                PushConfiguration pushConfiguration2 = this.mConfiguration;
                openHmsPush = false;
                if (pushConfiguration2 != null) {
                    openHmsPush = pushConfiguration2.getOpenFCMPush();
                }
                break;
            case 3:
                PushConfiguration pushConfiguration3 = this.mConfiguration;
                openHmsPush = false;
                if (pushConfiguration3 != null) {
                    openHmsPush = pushConfiguration3.getOpenCOSPush();
                    break;
                }
            case 4:
                PushConfiguration pushConfiguration4 = this.mConfiguration;
                if (pushConfiguration4 != null) {
                    openHmsPush = pushConfiguration4.getOpenFTOSPush();
                }
                break;
            default:
                openHmsPush = false;
                break;
        }
        return openHmsPush;
    }

    @Override // com.xiaomi.mipush.sdk.AbstractPushManager
    public void register() {
        MyLog.w("ASSEMBLE_PUSH : assemble push register");
        if (this.mManagers.size() <= 0) {
            initAssemblePushManager();
        }
        if (this.mManagers.size() > 0) {
            for (AbstractPushManager abstractPushManager : this.mManagers.values()) {
                if (abstractPushManager != null) {
                    abstractPushManager.register();
                }
            }
            AssemblePushHelper.checkAssemblePushStatus(this.mContext);
        }
    }

    public void removeManager(AssemblePush assemblePush) {
        this.mManagers.remove(assemblePush);
    }

    public void setConfiguration(PushConfiguration pushConfiguration) {
        this.mConfiguration = pushConfiguration;
        this.oldOCValue = OnlineConfig.getInstance(this.mContext).getBooleanValue(ConfigKey.AggregatePushSwitch.getValue(), true);
        if (this.mConfiguration.getOpenHmsPush() || this.mConfiguration.getOpenFCMPush() || this.mConfiguration.getOpenCOSPush()) {
            OnlineConfig.getInstance(this.mContext).addOCUpdateCallbacks(new OnlineConfig.OCUpdateCallback(101, "assemblePush") { // from class: com.xiaomi.mipush.sdk.AssemblePushCollectionsManager.1
                @Override // com.xiaomi.push.service.OnlineConfig.OCUpdateCallback
                protected void onCallback() {
                    boolean booleanValue = OnlineConfig.getInstance(AssemblePushCollectionsManager.this.mContext).getBooleanValue(ConfigKey.AggregatePushSwitch.getValue(), true);
                    if (AssemblePushCollectionsManager.this.oldOCValue != booleanValue) {
                        AssemblePushCollectionsManager.this.oldOCValue = booleanValue;
                        AssemblePushHelper.registerAssemblePush(AssemblePushCollectionsManager.this.mContext);
                    }
                }
            });
        }
    }

    @Override // com.xiaomi.mipush.sdk.AbstractPushManager
    public void unregister() {
        MyLog.w("ASSEMBLE_PUSH : assemble push unregister");
        for (AbstractPushManager abstractPushManager : this.mManagers.values()) {
            if (abstractPushManager != null) {
                abstractPushManager.unregister();
            }
        }
        this.mManagers.clear();
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/mipush/sdk/AssemblePushCollectionsManager.kt
package com.xiaomi.mipush.sdk

import com.xiaomi.channel.commonutils.logger.MyLog

import android.content.Context
import com.xiaomi.push.service.OnlineConfig
import com.xiaomi.xmpush.thrift.ConfigKey

class AssemblePushCollectionsManager private constructor(context: Context) : AbstractPushManager {
    companion object {
        @Volatile
        private var sInstance: AssemblePushCollectionsManager? = null

        @JvmStatic
        fun getInstance(context: Context): AssemblePushCollectionsManager {
            return sInstance ?: synchronized(this) {
                sInstance ?: AssemblePushCollectionsManager(context).also { sInstance = it }
            }
        }
    }

    private val mContext: Context = context.applicationContext
    private var mConfiguration: PushConfiguration? = null
    private val mManagers: MutableMap<AssemblePush, AbstractPushManager> = HashMap()
    private var oldOCValue = false

    private fun initAssemblePushManager() {
        val pushConfiguration = mConfiguration ?: return
        if (pushConfiguration.openHmsPush) {
            MyLog.w("ASSEMBLE_PUSH :  HW user switch : ${pushConfiguration.openHmsPush} HW online switch : ${AssemblePushHelper.isOpenAssemblePushOnlineSwitch(mContext, AssemblePush.ASSEMBLE_PUSH_HUAWEI)} HW isSupport : ${PhoneBrand.HUAWEI == AssemblePushUtils.getPhoneBrand(mContext)}")
        }
        if (pushConfiguration.openHmsPush && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(mContext, AssemblePush.ASSEMBLE_PUSH_HUAWEI) && PhoneBrand.HUAWEI == AssemblePushUtils.getPhoneBrand(mContext)) {
            if (!contain(AssemblePush.ASSEMBLE_PUSH_HUAWEI)) {
                PushManagerFactory.getManager(mContext, AssemblePush.ASSEMBLE_PUSH_HUAWEI)?.let { addManager(AssemblePush.ASSEMBLE_PUSH_HUAWEI, it) }
            }
            MyLog.v("hw manager add to list")
        } else if (contain(AssemblePush.ASSEMBLE_PUSH_HUAWEI)) {
            getManager(AssemblePush.ASSEMBLE_PUSH_HUAWEI)?.also { manager ->
                removeManager(AssemblePush.ASSEMBLE_PUSH_HUAWEI)
                manager.unregister()
            }
        }
        if (pushConfiguration.openFCMPush) {
            MyLog.w("ASSEMBLE_PUSH :  FCM user switch : ${pushConfiguration.openFCMPush} FCM online switch : ${AssemblePushHelper.isOpenAssemblePushOnlineSwitch(mContext, AssemblePush.ASSEMBLE_PUSH_FCM)} FCM isSupport : ${AssemblePushUtils.isGoogleServiceSatisfied(mContext)}")
        }
        if (pushConfiguration.openFCMPush && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(mContext, AssemblePush.ASSEMBLE_PUSH_FCM) && AssemblePushUtils.isGoogleServiceSatisfied(mContext)) {
            if (!contain(AssemblePush.ASSEMBLE_PUSH_FCM)) {
                PushManagerFactory.getManager(mContext, AssemblePush.ASSEMBLE_PUSH_FCM)?.let { addManager(AssemblePush.ASSEMBLE_PUSH_FCM, it) }
            }
            MyLog.v("fcm manager add to list")
        } else if (contain(AssemblePush.ASSEMBLE_PUSH_FCM)) {
            getManager(AssemblePush.ASSEMBLE_PUSH_FCM)?.also { manager ->
                removeManager(AssemblePush.ASSEMBLE_PUSH_FCM)
                manager.unregister()
            }
        }
        if (pushConfiguration.openCOSPush) {
            MyLog.w("ASSEMBLE_PUSH :  COS user switch : ${pushConfiguration.openCOSPush} COS online switch : ${AssemblePushHelper.isOpenAssemblePushOnlineSwitch(mContext, AssemblePush.ASSEMBLE_PUSH_COS)} COS isSupport : ${AssemblePushUtils.isColorOSPushSupport(mContext)}")
        }
        if (pushConfiguration.openCOSPush && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(mContext, AssemblePush.ASSEMBLE_PUSH_COS) && AssemblePushUtils.isColorOSPushSupport(mContext)) {
            PushManagerFactory.getManager(mContext, AssemblePush.ASSEMBLE_PUSH_COS)?.let { addManager(AssemblePush.ASSEMBLE_PUSH_COS, it) }
        } else if (contain(AssemblePush.ASSEMBLE_PUSH_COS)) {
            getManager(AssemblePush.ASSEMBLE_PUSH_COS)?.also { manager ->
                removeManager(AssemblePush.ASSEMBLE_PUSH_COS)
                manager.unregister()
            }
        }
        if (pushConfiguration.openFTOSPush && AssemblePushHelper.isOpenAssemblePushOnlineSwitch(mContext, AssemblePush.ASSEMBLE_PUSH_FTOS) && AssemblePushUtils.isFunTouchOSPushSupport(mContext)) {
            PushManagerFactory.getManager(mContext, AssemblePush.ASSEMBLE_PUSH_FTOS)?.let { addManager(AssemblePush.ASSEMBLE_PUSH_FTOS, it) }
        } else if (contain(AssemblePush.ASSEMBLE_PUSH_FTOS)) {
            getManager(AssemblePush.ASSEMBLE_PUSH_FTOS)?.also { manager ->
                removeManager(AssemblePush.ASSEMBLE_PUSH_FTOS)
                manager.unregister()
            }
        }
    }

    fun addManager(assemblePush: AssemblePush, abstractPushManager: AbstractPushManager) {
        if (abstractPushManager != null) {
            mManagers.remove(assemblePush)
            mManagers[assemblePush] = abstractPushManager
        }
    }

    fun contain(assemblePush: AssemblePush): Boolean = mManagers.containsKey(assemblePush)

    fun getManager(assemblePush: AssemblePush): AbstractPushManager? = mManagers[assemblePush]

    fun getUserSwitch(assemblePush: AssemblePush): Boolean {
        return when (assemblePush) {
            AssemblePush.ASSEMBLE_PUSH_HUAWEI -> mConfiguration?.openHmsPush ?: false
            AssemblePush.ASSEMBLE_PUSH_FCM -> mConfiguration?.openFCMPush ?: false
            AssemblePush.ASSEMBLE_PUSH_COS -> mConfiguration?.openCOSPush ?: false
            AssemblePush.ASSEMBLE_PUSH_FTOS -> mConfiguration?.openFTOSPush ?: false
        }
    }

    override fun register() {
        MyLog.w("ASSEMBLE_PUSH : assemble push register")
        if (mManagers.isEmpty()) {
            initAssemblePushManager()
        }
        if (mManagers.isNotEmpty()) {
            for (abstractPushManager in mManagers.values) {
                abstractPushManager?.register()
            }
            AssemblePushHelper.checkAssemblePushStatus(mContext)
        }
    }

    fun removeManager(assemblePush: AssemblePush) {
        mManagers.remove(assemblePush)
    }

    fun setConfiguration(pushConfiguration: PushConfiguration) {
        mConfiguration = pushConfiguration
        oldOCValue = OnlineConfig.getInstance(mContext).getBooleanValue(ConfigKey.AggregatePushSwitch.value, true)
        if (pushConfiguration.openHmsPush || pushConfiguration.openFCMPush || pushConfiguration.openCOSPush) {
            OnlineConfig.getInstance(mContext).addOCUpdateCallbacks(object : OnlineConfig.OCUpdateCallback(101, "assemblePush") {
                override fun onCallback() {
                    val booleanValue = OnlineConfig.getInstance(mContext).getBooleanValue(ConfigKey.AggregatePushSwitch.value, true)
                    if (this@AssemblePushCollectionsManager.oldOCValue != booleanValue) {
                        this@AssemblePushCollectionsManager.oldOCValue = booleanValue
                        AssemblePushHelper.registerAssemblePush(this@AssemblePushCollectionsManager.mContext)
                    }
                }
            })
        }
    }

    override fun unregister() {
        MyLog.w("ASSEMBLE_PUSH : assemble push unregister")
        for (abstractPushManager in mManagers.values) {
            abstractPushManager?.unregister()
        }
        mManagers.clear()
    }
}
